catalog/core/catalog-core-transform-impl/src/test/java/org/codice/ddf/catalog/locator/impl/TransformerLocatorImplTest.java

```diff
@@ -0,0 +1,366 @@
+    when(transformer.getMimeTypes()).thenReturn(new HashSet<>(Arrays.asList(mimeTypes)));
+    serviceMap.computeIfAbsent(clazz, aClass -> new ArrayList()).add(serviceReference);
+
+    return transformer;
+  }
+
+  private MultiInputTransformer mockMultiInputTransformer(String id) {
+    return mockMultiTransformer(MultiInputTransformer.class, id);
+  }
+
+  private <T extends TransformerProperties> T mockMultiTransformer(Class<T> clazz, String id) {
+    T transformer = mock(clazz);
+    ServiceReference<T> serviceReference = mock(ServiceReference.class);
+    when(bundleContext.getService(serviceReference)).thenReturn(transformer);
+    when(transformer.getId()).thenReturn(id);
+    when(serviceReference.getProperty(eq(Constants.SERVICE_ID))).thenReturn(id);
+
+    serviceMap.computeIfAbsent(clazz, aClass -> new ArrayList()).add(serviceReference);
+
+    return transformer;
+  }
+
+  private MultiMetacardTransformer mockMultiMetacardTransformer(MimeType... mimeTypes) {
+    return mockMultiTransformer(MultiMetacardTransformer.class, mimeTypes);
+  }
+
+  private MultiMetacardTransformer mockMultiMetacardTransformer(String id) {
+    return mockMultiTransformer(MultiMetacardTransformer.class, id);
+  }
+
+  private InputTransformer mockInputTransformer(Map<String, Object> serviceProperties) {
+    return mockTransformer(serviceProperties, InputTransformer.class);
+  }
+
+  private MetacardTransformer mockMetacardTransformer(Map<String, Object> serviceProperties) {
+    return mockTransformer(serviceProperties, MetacardTransformer.class);
+  }
+
+  private QueryResponseTransformer mockQueryResponseTransformer(
+      Map<String, Object> serviceProperties) {
+    return mockTransformer(serviceProperties, QueryResponseTransformer.class);
+  }
+
+  private <T> T mockTransformer(Map<String, Object> serviceProperties, Class<T> clazz) {
+
+    T metacardTransformer = mock(clazz);
+    ServiceReference<T> serviceReference = mock(ServiceReference.class);
+    when(serviceReference.getPropertyKeys())
+        .thenReturn(serviceProperties.keySet().toArray(new String[0]));
+    when(bundleContext.getService(serviceReference)).thenReturn(metacardTransformer);
+    when(serviceReference.getProperty(eq(TransformerProperties.MIME_TYPE)))
+        .thenReturn(
+            serviceProperties
+                .entrySet()
+                .stream()
+                .filter(
+                    stringObjectEntry ->
+                        stringObjectEntry.getKey().equals(TransformerProperties.MIME_TYPE))
+                .map(Map.Entry::getValue)
+                .filter(List.class::isInstance)
+                .map(List.class::cast)
+                .flatMap(Collection::stream)
+                .filter(String.class::isInstance)
+                .map(String.class::cast)
+                .collect(Collectors.toList()));
+    when(serviceReference.getProperty(eq(Constants.SERVICE_ID)))
+        .thenReturn(
+            serviceProperties
```

Comment:
Wouldn't `.thenReturn(serviceProperties.get(Constants.SERVICE_ID))` do the same thing?